presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/HiveMetastoreCommitHandle.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore;
+
+import com.facebook.presto.spi.connector.ConnectorCommitHandle;
+import org.joda.time.DateTime;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class HiveMetastoreCommitHandle
+        implements ConnectorCommitHandle
+{
+    private final Map<String, List<DateTime>> lastDataCommitTimes = new HashMap<>();
+
+    public void addLastDataCommitTimes(String key, List<DateTime> value)
+    {
+        if (!lastDataCommitTimes.containsKey(key)) {
+            lastDataCommitTimes.put(key, new ArrayList<DateTime>());
+        }
+        lastDataCommitTimes.get(key).addAll(value);
+    }
+
+    public boolean containLastDataCommitTimes(String key)
+    {
+        return lastDataCommitTimes.containsKey(key);
+    }
+
+    public Set<String> getLastDataCommitTimesKeys()
+    {
+        return lastDataCommitTimes.keySet();
+    }
+
+    public List<DateTime> getLastDataCommitTimes(String key)
+    {
+        return lastDataCommitTimes.get(key);
+    }
+
+    public void removeLastDataCommitTimes(String key)
+    {
+        lastDataCommitTimes.remove(key);
+    }
```

Comment:
The members of a handle class should be immutable. If you wanna add new stuff into a class, it's better to create a new one with copy.

Reply:
Good to know this! When you say a copy, do you mean the copy of the whole handle, or just the class member?

Reply:
copy of the whole handle,